nginx-controller/controller/controller.go

```diff
@@ -900,15 +940,23 @@ func (lbc *LoadBalancerController) createIngress(ing *extensions.Ingress) (*ngin
 
 	validRules := 0
 
-	for _, rule := range ing.Spec.Rules {
-		if rule.IngressRuleValue.HTTP == nil {
-			continue
-		}
+	mergeableIngress := false
```

Comment:
Had to add this flag to the createIngress() to support masters without a Path.

Reply:
Can we change the master Ingress resource before calling createIngress? We need to make sure that `rule.IngressRuleValue.HTTP` is not nil. This way we don't have to change the `createIngress` method and ` generateNginxCfg` in Configurator. What do you think?

Reply:
Sounds Good! We can set it to an empty object.